features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/DashDirectUserAuthFragment.kt

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2021 Gabor Varadi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dash.wallet.features.exploredash.ui
+
+import android.content.Context
+import android.os.Build
+import android.os.Bundle
+import android.util.Patterns
+import android.view.View
+import android.view.ViewGroup
+import android.view.WindowInsets
+import android.view.inputmethod.EditorInfo
+import android.view.inputmethod.InputMethodManager
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import androidx.annotation.StringRes
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.updateLayoutParams
+import androidx.core.widget.doOnTextChanged
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.navGraphViewModels
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.launch
+import org.dash.wallet.common.data.ResponseResource
+import org.dash.wallet.common.ui.viewBinding
+import org.dash.wallet.common.util.safeNavigate
+import org.dash.wallet.features.exploredash.R
+import org.dash.wallet.features.exploredash.databinding.FragmentDashDirectUserAuthBinding
+import org.slf4j.LoggerFactory
+
+@FlowPreview
+@ExperimentalCoroutinesApi
+@AndroidEntryPoint
+class DashDirectUserAuthFragment : Fragment(R.layout.fragment_dash_direct_user_auth) {
+    companion object {
+        private val log = LoggerFactory.getLogger(DashDirectUserAuthFragment::class.java)
+    }
+
+    private var currentDirectUserAuthType: DashDirectUserAuthType?=null
+    private val binding by viewBinding(FragmentDashDirectUserAuthBinding::bind)
+    private val exploreViewModel: ExploreViewModel by navGraphViewModels(R.id.explore_dash) { defaultViewModelProviderFactory }
```

Comment:
I thought we have a dash direct viewModel? Let's not put dash direct logic into the `ExploreViewModel` if we can avoid it.

Reply:
I need that part to check and map user email in the search fragment